ext/ddtrace_profiling_native_extension/collectors_discrete_dynamic_sampler.c

```diff
@@ -13,12 +13,27 @@
 
 #define EMA_SMOOTHING_FACTOR 0.6
 
-void discrete_dynamic_sampler_init(discrete_dynamic_sampler *sampler, const char *debug_name) {
+#define ERR_CLOCK_FAIL "failed to get clock time"
+#define ERR_TEST "test failure"
+
+static const char* _discrete_dynamic_sampler_set_overhead_target_percentage(discrete_dynamic_sampler *sampler, double target_overhead, long now_ns);
+
+static const char* _discrete_dynamic_sampler_init(discrete_dynamic_sampler *sampler, const char *debug_name, long now_ns) {
   sampler->debug_name = debug_name;
-  discrete_dynamic_sampler_set_overhead_target_percentage(sampler, BASE_OVERHEAD_PCT);
+  return _discrete_dynamic_sampler_set_overhead_target_percentage(sampler, BASE_OVERHEAD_PCT, now_ns);
+}
+
+const char* discrete_dynamic_sampler_init(discrete_dynamic_sampler *sampler, const char *debug_name) {
+  long now = monotonic_wall_time_now_ns(DO_NOT_RAISE_ON_FAILURE);
+  return _discrete_dynamic_sampler_init(sampler, debug_name, now);
 }
 
-static void _discrete_dynamic_sampler_reset(discrete_dynamic_sampler *sampler, long now_ns) {
+static const char* _discrete_dynamic_sampler_reset(discrete_dynamic_sampler *sampler, long now_ns) {
+  if (now_ns <= 0) {
+    sampler->error = ERR_CLOCK_FAIL;
+    return sampler->error;
+  }
```

Comment:
```suggestion
  if (now_ns <= 0) return sampler->error = ERR_CLOCK_FAIL;
```

(I'll show myself out :rofl:)